.github/workflows/ci.yml

```diff
@@ -0,0 +1,117 @@
+name: CI Checks
+
+on:
+  push:
+    branches: ["**"]
+  pull_request:
+    branches: [master]
+  workflow_dispatch:
+
+jobs:
+  demo-build:
```

Comment:
demo-test-build? as this builds both demos and integration tests?

Reply:
Renamed to build-check with a step for demos and a step for system test.